hazelcast/src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -78,11 +79,15 @@
     private boolean isEmptyAddressListAlreadyLogged;
 
     AwsDiscoveryStrategy(Map<String, Comparable> properties) {
-        super(LOGGER, properties);
+        this(null, properties);
+    }
 
-        AwsConfig awsConfig = createAwsConfig();
+    AwsDiscoveryStrategy(DiscoveryNode discoveryNode, Map<String, Comparable> properties) {
+        super(LOGGER, properties);
+        AwsConfig awsConfig = createAwsConfig()
+                .setDiscoveryMode(discoveryNode == null ? DiscoveryMode.Client : DiscoveryMode.Member)
```

Comment:
So Hz is not creating a DiscoveryNode object if it is a client?

Reply:
Yeap, that's right. Here is from the Javadoc of `DiscoveryStrategyFactory#newDiscoveryStrategy`
```
* @param discoveryNode the current local <code>DiscoveryNode</code>, representing the local
*                      connection information if running on a Hazelcast member, otherwise on
*                      Hazelcast clients always <code>null</code>
```